src/env.cc

```diff
@@ -1051,6 +1052,25 @@ void Environment::AddArrayBufferAllocatorToKeepAliveUntilIsolateDispose(
   keep_alive_allocators_->insert(allocator);
 }
 
+bool Environment::RunWeakRefCleanup() {
+  isolate()->ClearKeptObjects();
+
+  TryCatchScope try_catch(this);
```

Comment:
Kind of a micro-optimization but preceding this with a `if (cleanup_finalization_groups_.empty()) return true;` and changing the while to a do/while should be marginally faster because it doesn't create the `TryCatchScope` in the (what I presume is) the common case.